packages/nuxt/src/app/composables/fetch.ts

```diff
@@ -44,7 +46,7 @@ export function useFetch<
   arg2?: string
 ) {
   const [opts = {}, autoKey] = typeof arg1 === 'string' ? [{}, arg1] : [arg1, arg2]
-  const _key = opts.key || autoKey
+  const _key = opts.key || (typeof request === 'string' ? hash(withBase(unref(opts.baseURL) || '', withQuery(request as string, unref(opts.params) || {}))) : autoKey)
```

Comment:
(didn't locally tested) We can pass raw object to hash function. I guess also withBase not needed we mainly need digest.

```suggestion
  const _reqHash = typeof request === 'string' ? hash([unref(opts.baseURL)]) : ''
  const _key = opts.key || (autoKey + '-' + (typeof request === 'string' ? hash([request, unref(opts.baseURL), unref(opts.params)]) : ''))
```

Reply:
updated. though seems like this approach would prevent sharing useFetch state between pages?